include/cliffordsynthesis/encoding/GateEncoder.hpp

```diff
@@ -30,18 +30,24 @@
     logicbase::LogicMatrix3D gS{};
     // variables for the two-qubit gates
     logicbase::LogicMatrix3D gC{};
+    // variables for the pauli-qubit gates
+    logicbase::LogicMatrix gP{};
```

Comment:
These variables are only used by the TwoQubitGateEncoder, please move the method there.